### PR TITLE
[deps]: pin faraday at >= 1.0 and bump version

### PR DESCRIPTION
**Pull request by jongirard**
In this PR I updated the faraday dependency to `~> 1.0` as previously it was pinned to an older version and causing compatibility issues in my Gemfile with other gems.

### CODE REVIEW

**Comment by MatteoJoliveau** [comment]
Hi @jongirard , thank you for this PR!
Since this is a major upgrade to Faraday, and to a 1.0 version no less, I wanna double check that the upgrade does not break anything for the end-users of the library.

Since the current testing infrastructure is a bit broken at the moment, I'll put this PR on hold. We'll get back to it as soon as we have proper pipelines back up 👍🏻

**Comment by jongirard** [comment]
Sounds good 👍🏻 I can report that I have been using it at ~1.0 for a little while now without any issues so hopefully the tests go smoothly

**Comment by dmitry** [comment]
@MatteoJoliveau any help required on this? In other case there are no possibility to upgrade faraday for the other gems, means other gems cannot be upgraded too.

**Comment by MatteoJoliveau** [comment]
@dmitry we're working on it. We're fixing the CI pipelines to have PR checks back. Bear with us 🙏🏻 
In the meantime @jongirard can I as you to change the version to `0.5.0` instead of `0.4.1` since it's not a bugfix?

File: lib/onesignal/version.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
 module OneSignal
-  VERSION = '0.4.0'
+  VERSION = '0.4.1'
```
Comment (MatteoJoliveau) [inline lib/onesignal/version.rb:4]:
```suggestion
  VERSION = '0.5.0'
```

**Review by MatteoJoliveau — CHANGES_REQUESTED** [review]
Minor change to properly follow semver 👍🏻

**Pull request merged**